packages/web-components/src/components/va-promo-banner/va-promo-banner.tsx

```diff
@@ -0,0 +1,156 @@
+  shadow: true,
+})
+export class VaPromoBanner {
+  @Element() el: HTMLElement;
+
+  /**
+   * `href` attribute for the `<a>` tag.
+   */
+  @Prop() href: string;
+
+  /**
+   * `target` attribute for the `<a>` tag.
+   */
+  @Prop() target: string;
+
+  /**
+   * Controls which icon gets used. Options are 'announcement', 'news', or 'email-signup'.
+   * */
+  @Prop() type: 'announcement' | 'news' | 'email-signup';
+
+  /**
+   * Analytics tracking function(s) will not be called
+   */
+  @Prop() disableAnalytics: boolean;
+
+  /**
+   * Fires when the component is closed by clicking on the close icon.
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  closeEvent: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link or the dismiss icon is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * Keep track of locally dismissed Banners
+   * */
+  @State() dismissedBanners: Array<String> = [];
+
+  private closeHandler(): void {
+    this.closeEvent.emit();
+    // Derive the updated dismissed banners based on the id attribute set on the component.
+    const updatedDismissedBanners = [...this.dismissedBanners, this.el.id];
+
+    // Update dismissedBanners in state.
+    this.dismissedBanners = updatedDismissedBanners;
+
+    // Add the promo banner ID to local storage.
+    localStorage.setItem(
+      DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY,
+      JSON.stringify(updatedDismissedBanners),
+    );
+  }
+
+  private handleLinkClick(e: MouseEvent): void {
+    if (!this.disableAnalytics) {
+      const targetEl = e.target as HTMLElement;
+      const targetAnchor =
+        targetEl?.tagName === 'A' ||
+        (targetEl?.tagName === 'SLOT' &&
+          targetEl?.parentElement.tagName === 'A');
+      // If it's a link being clicked, dispatch an analytics event
+      if (targetAnchor) {
+        const detail = {
+          componentName: 'va-promo-banner',
+          action: 'linkClick',
+          details: {
+            text:
+              targetEl.tagName === 'A' ? targetEl.innerText : this.el.innerText,
+            href: this.href,
+            target: this.target,
+            type: this.type,
+          },
+        };
+        this.componentLibraryAnalytics.emit(detail);
+      }
+    }
+  }
+
+  componentWillLoad() {
+    // Derive dismissed banners from storage.
+    const dismissedBannersString = localStorage.getItem(
+      DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY,
+    );
+    this.dismissedBanners = dismissedBannersString
+      ? JSON.parse(dismissedBannersString)
+      : [];
+  }
+
+  render() {
+    // Derive if the banner is dismissed based on the id attribute set on the component.
+    const isBannerDismissed = this.dismissedBanners?.includes(this.el.id);
```

Comment:
If `id` is necessary in order for dismiss functionality to work it may be good to have a test to show that dismiss _won't_ work without an `id`. In the React version we left those implementation details up to the consumer, so now that we're internalizing it having a test case describing the intentional behavior could act as a form of documentation.

Reply:
@bkjohnson added a test for this and also converted the css to flex instead of table